controllers/dependencies.go

```diff
@@ -96,11 +102,51 @@ func (r *OperatorPipelineReconciler) applyManifests(fileName string, Namespace s
 		return err
 	}
 
-	obj.SetNamespace(Namespace)
-	if err = r.Client.Create(context.Background(), obj); err != nil {
-		log.Error(err, "Couldn't create resource")
+	obj.SetNamespace(meta.Namespace)
+	err = r.Get(context.Background(), types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace}, obj)
+	if err != nil && errors.IsNotFound(err) {
+
+		if err = r.Client.Create(context.Background(), obj); err != nil {
+			log.Error(err, "failed to create pipeline resource")
+			return err
+		}
+	} else {
```

Comment:
I believe this has to be more specific:
```
else if (err != nil && errors.IsAlreadyExist(err) {
   // update the resource
} else {
  return err
}
```

Reply:
The `else` is for the `Get` not for the `Create` so if the resource exist in the cluster `obj` is populated and error is `nil`. You would be correct if it was on the `Create` but I felt that code was harder to read and overkill.